hazelcast/src/main/java/com/hazelcast/map/impl/proxy/NearCachedMapProxyImpl.java

```diff
@@ -262,6 +262,12 @@ protected Data removeInternal(Data key) {
     }
 
     @Override
+    protected void removeAllInternal(Predicate predicate) {
+        super.removeAllInternal(predicate);
+        nearCache.clear();
```

Comment:
Here are the options we discussed with @ahmetmircik :
(1) execute near cache full scan asynchronously to locate entries matching the predicate and remove them. Pros: accurate, does not throw away valid items from your near cache, Cons: may cause stale reads until the async full scan is done - so it’s different from `remove`‘s behavior
(2) return the removed keys from the `EntryProcessor` and only remove these from the near cache. Pros: as above, Cons: additional serialization/network costs, places burden on opposite use case in which user’s predicate matches a large number of keys
(3) make it “adaptive”: the `removeAll` operations return the number of keys removed; on the local member, if number of entries removed > (some threshold percentage) then just clear all otherwise do the full scan & check against the predicate. Making this threshold overridable by a property would allow users to tune it according to their use case (a small threshold to avoid full near cache scan and quickly turn to full `clear` at the cost of emptying near cache or a large threshold that would increase latency of `removeAll` and keep their near cache warm).

Of the options above, I am leaning more towards (3) but it may pose some implementation challenges.